dashboard/api_edx_cache_test.py

```diff
@@ -38,6 +38,66 @@ class CachedEdxUserDataTests(ESTestCase):
 
     @classmethod
     def setUpTestData(cls):
+        cls.user = UserFactory.create()
+        # Create Programs, Courses, CourseRuns...
+        cls.p1_course_run = CourseRunFactory.create(edx_course_key='p1_course_run')
+        p2 = ProgramFactory.create(full=True)
+        first_course = p2.course_set.first()
+        extra_course = CourseFactory.create(program=p2)
+        cls.p2_course_run_1 = CourseRunFactory.create(course=first_course, edx_course_key='p2_course_run_1')
+        cls.p2_course_run_2 = CourseRunFactory.create(course=extra_course, edx_course_key='p2_course_run_2')
+        # Create cached edX data
+        enrollment1 = CachedEnrollmentFactory.create(user=cls.user, course_run=cls.p1_course_run)
+        enrollment2 = CachedEnrollmentFactory.create(user=cls.user, course_run=cls.p2_course_run_1)
+        enrollment3 = CachedEnrollmentFactory.create(user=cls.user, course_run=cls.p2_course_run_2)
+        certificate1 = CachedCertificateFactory.create(user=cls.user, course_run=cls.p1_course_run)
+        certificate2 = CachedCertificateFactory.create(user=cls.user, course_run=cls.p2_course_run_1)
+        certificate3 = CachedCertificateFactory.create(user=cls.user, course_run=cls.p2_course_run_2)
+        curgrade1 = CachedCurrentGradeFactory.create(user=cls.user, course_run=cls.p1_course_run)
+        curgrade2 = CachedCurrentGradeFactory.create(user=cls.user, course_run=cls.p2_course_run_1)
+        curgrade3 = CachedCurrentGradeFactory.create(user=cls.user, course_run=cls.p2_course_run_2)
+        cls.enrollments = [enrollment1, enrollment2, enrollment3]
+        cls.certificates = [certificate1, certificate2, certificate3]
+        cls.current_grades = [curgrade1, curgrade2, curgrade3]
+
+    def assert_edx_data_has_given_ids(self, edx_user_data, ids):
+        """Asserts that all edX object course id sets match the given list of ids"""
+        assert sorted(edx_user_data.enrollments.get_enrolled_course_ids()) == ids
+        assert sorted(edx_user_data.certificates.all_courses_verified_certs) == ids
+        assert sorted(edx_user_data.current_grades.all_course_ids) == ids
+
+    def test_edx_data_fetch_and_set(self):
+        """Test that a user's edX data is properly fetched and set onto object properties"""
+        edx_user_data = CachedEdxUserData(self.user)
+        assert isinstance(edx_user_data.enrollments, Enrollments)
+        assert isinstance(edx_user_data.certificates, Certificates)
+        assert isinstance(edx_user_data.current_grades, CurrentGrades)
+        self.assert_edx_data_has_given_ids(edx_user_data, ['p1_course_run', 'p2_course_run_1', 'p2_course_run_2'])
```

Comment:
please move the ids to `cls` variables and reference them both here and during the creation of the runs